apps/workspaces/permissions.py

```
@@ -30,3 +32,20 @@
+class IsAuthenticatedForInternalAPI(permissions.BasePermission):
+    """
+    Custom auth for internal APIs
+    """
+    def has_permission(self, request, view):
+        # Client sends a token in the header, which we decrypt and compare with the Client Secret
+        cipher_suite = Fernet(settings.ENCRYPTION_KEY)
+        try:
+            decrypted_password = cipher_suite.decrypt(request.headers['X-Internal-API-Client-ID'].encode('utf-8')).decode('utf-8')
+            if decrypted_password == settings.E2E_TESTS_CLIENT_SECRET:
+                return True
+        except:
```

Ruff (0.7.0)

Do not use bare `except`

(E722)